presto-main/src/main/java/com/facebook/presto/server/ServerInfoResource.java

```diff
@@ -108,7 +111,12 @@ public NodeState getServerState()
             return SHUTTING_DOWN;
         }
         else {
-            return ACTIVE;
+            if (nodeResourceStatusProvider.hasResources()) {
```

Comment:
this should be only applicable for coordinator. So a coordinator check is needed. For worker, nodeResourceStatusProvider won't be available.

Reply:
missed worked implementation. Resolved.